client/src/modules/cohort/components/CohortHeader.jsx

```diff
@@ -36,17 +36,19 @@ class CohortHeader extends PureComponent {
     const {
       isDescriptionTextareaVisible,
       isNameInputVisible,
-      name
+      nameInputValue
     } = this.state;
 
     if (isDescriptionTextareaVisible && isClickedOutside) {
-      this.setState({ isDescriptionTextareaVisible: false });
       this.handleDescriptionUpdate();
       return;
     }
 
-    if (isNameInputVisible && name.trim().length >= 1 && isClickedOutside) {
-      this.setState({ isNameInputVisible: false });
+    if (
+      isNameInputVisible &&
+      nameInputValue.trim().length >= 1 &&
```

Comment:
please use lodash _trim everywhere instead of native .trim() method. 
Let's make it consistent across whole file.